File: fc-chat/demo/part3/ch3/03-2-message-system-add-channel/src/main/java/net/prostars/messagesystem/dto/websocket/outbound/DisconnectResponse.java
package net.prostars.messagesystem.dto.websocket.outbound;

import net.prostars.messagesystem.constant.MessageType;
import net.prostars.messagesystem.constant.UserConnectionStatus;

public class DisconnectResponse extends BaseMessage {

  private final String username;
  private final UserConnectionStatus status;

  public DisconnectResponse(String username, UserConnectionStatus status) {
    super(MessageType.DISCONNECT_RESPONSE);
    this.username = username;
    this.status = status;
  }

  public String getUsername() {
    return username;
  }

  public UserConnectionStatus getStatus() {
    return status;
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/websocket/outbound/DisconnectResponse.kt
package com.yscorp.withpush.messagesystem.dto.websocket.outbound

import com.yscorp.withpush.messagesystem.constant.MessageType
import com.yscorp.withpush.messagesystem.constant.UserConnectionStatus


class DisconnectResponse(val username: String, val status: UserConnectionStatus) :
    BaseMessage(MessageType.DISCONNECT_RESPONSE) {
}
